roadmap/implementors-guide/src/types/availability.md

```diff
@@ -0,0 +1,29 @@
+# Availability
+
+One of the key roles of validators is to ensure availability of all data necessary to validate
+candidates for the duration of a challenge period. This is done via an erasure-coding of the data to keep available.
+
+## Signed Availability Bitfield
+
+A bitfield signed by a particular validator about the availability of pending candidates.
+
+
+```rust
+struct SignedAvailabilityBitfield {
+  validator_index: ValidatorIndex,
+  bitfield: Bitvec,
+  signature: ValidatorSignature,
+}
+
+struct Bitfields(Vec<(SignedAvailabilityBitfield)>), // bitfields sorted by validator index, ascending
+```
+
+The signed payload is the SCALE encoding of the tuple `(bitfield, signing_context)` where `signing_context` is a [`SigningContext`](../types/candidate.html#signing-context).
```

Comment:
I like what you've written about the nature of the signed payloads, here and elsewhere. It's outside the scope of this PR, but I think that it might make our life easier if we extracted that behavior entirely with something like this:

```rust
struct Signed<T> {
    payload: T,
    validator_index: ValidatorIndex,
    signature: ValidatorSignature,
}

impl<T: Encode> Signed<T> {
    fn sign(payload: T, context: SigningContext, index: ValidatorIndex, key: PrivateKey) -> Signed<T> { ... }
    fn validate(&self, context: SigningContext, key: PublicKey) -> bool { ... }
}
```

Reply:
Extracted into https://github.com/paritytech/polkadot/issues/1282 so that we don't lose this idea.